app/models/rule_group_rule.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+# Join table with the has-many-belongs-to-many relations between
+# RuleGroup and Rule
+class RuleGroupRule < ApplicationRecord
+  # These keep track of the parent-child relationship between
+  # rules and rule_groups
+  belongs_to :rule, optional: true
+  belongs_to :rule_group, optional: true
+  # These keep track of required/conflicting rules and rule groups
+  belongs_to :including, polymorphic: true, optional: true
+  belongs_to :requiring, polymorphic: true, optional: true
+  belongs_to :conflicting, polymorphic: true, optional: true
+  # These are needed to do ::RuleGroupRule.joins(rule/rule_group: :benchmark)
+  belongs_to :included_rule, class_name: 'Rule', foreign_key: 'including_id',
+                             inverse_of: :included_rules, optional: true
+  belongs_to :included_rule_group, class_name: 'RuleGroup', foreign_key: 'including_id',
+                                   inverse_of: :included_rule_groups, optional: true
+
+  validates :including_id, allow_nil: true, uniqueness: { scope: %i[requiring_id] }
+  validates :including_id, allow_nil: true, uniqueness: { scope: %i[conflicting_id] }
```

Comment:
Can't be these squashed into one? I have a feeling that these two are conflicting with each other.

Reply:
Also I'd rather validate against `including`, that's more semantically correct.

Reply:
Now that I think of it, I do think that they can be combined into one because it should still check for the uniqueness correctly, even if requiring_id or conflicting_id is nil

Reply:
https://guides.rubyonrails.org/active_record_validations.html#allow-blank